src/marqo/tensor_search/utils.py

```diff
@@ -349,3 +349,27 @@ def is_tensor_field(field: str,
         return field in tensor_fields
     else:
         return field not in non_tensor_fields
+
+
+def calculate_health_status(marqo_os_health_check_response: Optional[Dict]) -> dict:
+    """Calculate the health status of Marqo based on the response API from Marqo-os ."""
```

Comment:
This function actually has 2 purposes:
1. To handle the OpenSearch health check response 
2. To handle calculating the Marqo health

Perhaps we should break this into more functions: 
i. One for calculating the Marqo-os healthcheck, and handling errors from Marqo-os
ii. One for calculating this instance's health 
iii. One for calculating the aggregated system health
iv. One for formatting the health response for the user 

We can probably combine ii and iii into a single function for now